### PR TITLE
refs #1236 - Fixing multiple redirects handling using SlimerJS engine

### PR DESCRIPTION
**Pull request by ngotchac**
Fixing the strange redirect behavior when using SlimerJS' engine.

To run the specific test:

``` bash
bin/casperjs selftest --engine=slimerjs tests/suites/redirect.js
```

This will cause CasperJS to hang before using the fix

### CODE REVIEW

**Comment by istr** [comment]
Looks good. Complete PR including test -- obviously read CONTRIBUTING.md (and thanks for fixing the glitch in it).

However, might be better not to create a second server but to integrate the special redirected URL handling into the already existing, custom testing server ```tests/selftest.js```.

Your opinion @paazmaya, @hexid?

**Comment by paazmaya** [comment]
After checking the `tests/selftest.js`, I agree with @istr. Thank you for pointing that out.

File: tests/suites/redirect.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+var fs = require('fs');
+
+casper.test.begin('HTTP Redirect handling', 1, {
+    setUp: function() {
+        var pageFile = fs.pathJoin(phantom.casperPath, 'tests/site/redirect.html');
+
+        this.server = require('webserver').create();
+        this.server.listen(8090, function (request, response) {
+            if (request.url === '/redirect') {
```
Comment (istr) [inline tests/suites/redirect.js]: Could you please refactor this block into ```tests/selftest.js```...

File: tests/suites/redirect.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    tearDown: function() {
+        this.server.close();
+    },
+
+    test: function() {
+        casper.start();
+
+        // Redirect
+        casper.thenOpen('http://localhost:8090/redirect', function() {
```
Comment (istr) [inline tests/suites/redirect.js]: ...and use the standard self-test server here?

File: tests/suites/redirect.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            this.test.assert(true, 'Assert that the page has been fully loaded');
+        });
```
Comment (istr) [inline tests/suites/redirect.js]: Would be nice to check for status code and redirection target, too.

**Comment by istr** [comment]
Good candidate for cloned PR imho @BIGjuevos @mickaelandrieu

**Comment by BIGjuevos** [comment]
Agreed, self assigned unless @mickaelandrieu wants it.

**Comment by mickaelandrieu** [comment]
@BIGjuevos go for it, I'm too busy until the next release of PrestaShop :cat:

**Comment by kensoh** [comment]
Archiving PR using **ARCHIVE** label as CasperJS, like PhantomJS, is not actively maintained for now. These PRs are from 1-5 years ago but there just hasn't been maintainers to continue the work.

**Pull request closed without merging**